src/actions/actions.ts

```diff
@@ -4980,7 +4990,35 @@ class ActionDeleteLineVisualMode extends BaseCommand {
   keys = ["X"];
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
-    return await new DeleteOperator(this.multicursorIndex).run(vimState, position.getLineBegin(), position.getLineEnd());
+    if (vimState.currentMode === ModeName.Visual) {
+      return await new DeleteOperator(this.multicursorIndex).run(vimState,
+                                                                vimState.lastVisualSelectionStart.getLineBegin(),
```

Comment:
vimState.cursorStartPosition and vimState.cursorPosition. Sorry I didn't make that more clear and you had to hunt around.